R/readLPJmLClimateInput.R

```diff
@@ -0,0 +1,197 @@
+      close(filedata)
+
+    } else {
+      stop("File format of LPJmLClimateInput data unknown. Please provide .clm file format.")
+    }
+
+    if (subset == "wetDaysMonth") {
+
+      if (subtype$variable != "precipitation") stop("Subset 'wetDaysMonth' is only
+                                                   available for 'precipitation'")
+      x <- lpjclass::read.LPJ_input(file_name   = filename,
+                                    out_years   = paste0("y", years),
+                                    namesum     = TRUE,
+                                    ncells      = 67420,
+                                    rule4binary = ">0") / noAnnualPredictions
+
+      class(x) <- "array"
+      x        <- collapseNames(as.magpie(x, spatial = 1))
+
+
+    } else if (subset == "annualMean") {
+
+      x <- lpjclass::read.LPJ_input(file_name   = filename,
+                                    out_years   = paste0("y", years),
+                                    namesum     = TRUE,
+                                    ncells      = 67420) / noAnnualPredictions
+
+      class(x) <- "array"
+      x        <- collapseNames(as.magpie(x, spatial = 1))
+
+
+    } else if (subset == "annualSum") {
+
+      x <- lpjclass::read.LPJ_input(file_name   = filename,
+                                    out_years   = paste0("y", years),
+                                    namesum     = TRUE,
+                                    ncells      = 67420)
+
+      class(x) <- "array"
+      x        <- collapseNames(as.magpie(x, spatial = 1))
+
+
+    } else if (subset %in% c("monthlyMean", "monthlySum")) {
+      # define year sets (cut it in bunches)
+      bunchLength <- 1
+      yearsets    <- split(years, ceiling(seq_along(years) / bunchLength))
+
+      # define month mapping
+      monthLength <- c(jan = 31, feb = 28, mar = 31, apr = 30,
+                       may = 31, jun = 30, jul = 31, aug = 31,
+                       sep = 30, oct = 31, nov = 30, dec = 31)
+      daysMonth   <- NULL
+      for (m in 1:12) {
+        daysMonth <- c(daysMonth, rep(names(monthLength[m]),
+                                      monthLength[m]))
+      }
+      month2day   <- cbind(day   = 1:sum(monthLength),
+                           month = daysMonth)
+      monthLength <- as.magpie(monthLength)
+
+      # create output object
+      x <- NULL
+
+      # loop over bunches
+      for (b in seq_along(yearsets)) {
+        # read in a bunch of years
+        tmp <- lpjclass::read.LPJ_input(file_name = filename,
+                                        out_years = paste0("y", yearsets[[b]]),
+                                        namesum   = FALSE,
+                                        ncells    = 67420)
+        class(tmp) <- "array"
+        # KRISTINE: Please double-check whether following line makes sense
+        # (introduced because as.magpie doesn't work on 4-dimensional object)
+        tmp        <- abind::adrop(tmp, drop = 4)
```

Comment:
This is the only place where abind is needed. Since so many packages depend on mrcommons it would be great if no new dependencies are introduced unless absolutely necessary